### PR TITLE
build(deps): bump cosmos-sdk to v047

### PR DESCRIPTION
**Pull request by atheeshp**
Closes: #686

### CODE REVIEW

File: x/did/module.go [hunk 1]
```diff
@@ -129,17 +129,17 @@ func (am AppModule) Name() string {
...
 // Route returns the cheqd module's message routing key.
-func (am AppModule) Route() sdk.Route {
-	return sdk.NewRoute(types.RouterKey, keeper.NewHandler(am.keeper))
-}
+// func (am AppModule) Route() sdk.Route {
+// 	return sdk.NewRoute(types.RouterKey, keeper.NewHandler(am.keeper))
+// }
 
 // QuerierRoute returns the cheqd module's query routing key.
 func (AppModule) QuerierRoute() string { return types.QuerierRoute }
```
Comment (akhilkumarpilli) [inline x/did/module.go:137]:
```suggestion
```

File: x/resource/module.go [hunk 2]
```diff
@@ -134,18 +134,18 @@ func (am AppModule) Name() string {
...
-func (am AppModule) Route() sdk.Route {
-	return sdk.NewRoute(types.RouterKey, NewHandler(am.keeper, am.didKeeper))
-}
+// // Route returns the resource module's message routing key.
+// func (am AppModule) Route() sdk.Route {
+// 	return sdk.NewRoute(types.RouterKey, NewHandler(am.keeper, am.didKeeper))
+// }
 
 // QuerierRoute returns the resource module's query routing key.
 func (AppModule) QuerierRoute() string { return types.QuerierRoute }
```
Comment (akhilkumarpilli) [inline x/resource/module.go:143]: ditto

**Review by akhilkumarpilli — COMMENTED** [review]
Remove commented code if not needed

**Comment by akhilkumarpilli** [comment]
May be we need to add this code to provide newer gRPC services, have a look on this: https://github.com/cosmos/cosmos-sdk/blob/v0.47.8/UPGRADING.md#app-wiring

File: app/export.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ func (app *App) ExportAppStateAndValidators(
 		app.prepForZeroHeightGenesis(ctx, jailAllowedAddrs)
 	}
 
-	genState := app.mm.ExportGenesis(ctx, app.appCodec)
+	genState := app.ModuleManager.ExportGenesis(ctx, app.appCodec)
```
Comment (akhilkumarpilli) [inline app/export.go:33]:
```suggestion
	genState := app.ModuleManager.ExportGenesisForModules(ctx, app.appCodec, modulesToExport)
```

File: cmd/cheqd-noded/cmd/root.go [hunk 4]
```diff
@@ -89,27 +82,35 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 
 	rootCmd.AddCommand(
 		extendInit(genutilcli.InitCmd(app.ModuleBasics, app.DefaultNodeHome)),
-		genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, app.DefaultNodeHome),
-		genutilcli.MigrateGenesisCmd(),
-		genutilcli.GenTxCmd(app.ModuleBasics, encodingConfig.TxConfig, banktypes.GenesisBalancesIterator{}, app.DefaultNodeHome),
-		genutilcli.ValidateGenesisCmd(app.ModuleBasics),
 		AddGenesisAccountCmd(app.DefaultNodeHome),
```
Comment (akhilkumarpilli) [inline cmd/cheqd-noded/cmd/root.go:85]:
```suggestion
```
We can delete genaccounts.go file

File: make/proto.mk [hunk 5]
```diff
@@ -4,38 +4,40 @@
...
-containerProtoGenSwagger=cheqd-node-proto-gen-swagger-$(containerProtoVer)
-containerPulsar=cheqd-node-pulsar-gen-$(containerProtoVer)
+DOCKER := $(shell which docker)
+containerProtoVer=0.14.0
+containerProtoImage=ghcr.io/cosmos/proto-builder:$(containerProtoVer)
+protoImage=$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace $(containerProtoImage)
+# containerProtoGen=cheqd-node-proto-gen-$(containerProtoVer)
+# containerProtoFmt=cheqd-node-proto-fmt-$(containerProtoVer)
+# containerProtoGenSwagger=cheqd-node-proto-gen-swagger-$(containerProtoVer)
+# containerPulsar=cheqd-node-pulsar-gen-$(containerProtoVer)
```
Comment (Eengineer1) [inline make/proto.mk:14]: Is any further action required for swagger, pulsar generation included in any the above old logic?

Looks like the proto-builder is sufficient, but let's make sure that's the case.

If strong yes, let's remove these lines.

File: tests/upgrade/integration/v1/cli/query_did.go [hunk 6]
```diff
@@ -3,23 +3,23 @@ package cli
...
+// 	var resp didtypesv1.QueryGetDidResponse
+// 	err = integrationhelpers.Codec.UnmarshalJSON([]byte(res), &resp)
+// 	if err != nil {
+// 		return didtypesv1.QueryGetDidResponse{}, err
+// 	}
 
-	return resp, nil
-}
+// 	return resp, nil
+// }
```
Comment (Eengineer1) [inline tests/upgrade/integration/v1/cli/query_did.go:22]: Marking to remove commented out lines.

File: tests/upgrade/integration/v1/cli/query_resource.go [hunk 7]
```diff
@@ -3,23 +3,23 @@ package cli
...
+// 	var resp resourcetypesv1.QueryResourceResponse
+// 	err = integrationhelpers.Codec.UnmarshalJSON([]byte(res), &resp)
+// 	if err != nil {
+// 		return resourcetypesv1.QueryResourceResponse{}, err
+// 	}
 
-	return resp, nil
-}
+// 	return resp, nil
+// }
```
Comment (Eengineer1) [inline tests/upgrade/integration/v1/cli/query_resource.go:22]: Marking to remove commented out lines.

File: tests/upgrade/integration/v1/cli/tx_did.go [hunk 8]
```diff
@@ -1,34 +1,34 @@
...
+// 		args = append(args, base64.StdEncoding.EncodeToString(signInput.PrivKey))
+// 	}
 
-	args = append(args, GasParams...)
+// 	args = append(args, GasParams...)
 
-	return Tx(container, CliBinaryName, "cheqd", "create-did", OperatorAccounts[container], args...)
-}
+// 	return Tx(container, CliBinaryName, "cheqd", "create-did", OperatorAccounts[container], args...)
+// }
```
Comment (Eengineer1) [inline tests/upgrade/integration/v1/cli/tx_did.go:31]: Marking to remove commented out lines.

File: tests/upgrade/integration/v2/loader.go [hunk 9]
```diff
@@ -7,9 +7,10 @@ import (
 	integrationhelpers "github.com/cheqd/cheqd-node/tests/integration/helpers"
 	"github.com/cheqd/cheqd-node/x/did/client/cli"
 	didtypesv2 "github.com/cheqd/cheqd-node/x/did/types"
-	didtypesv1 "github.com/cheqd/cheqd-node/x/did/types/v1"
+
+	// didtypesv1 "github.com/cheqd/cheqd-node/x/did/types/v1"
 	resourcetypesv2 "github.com/cheqd/cheqd-node/x/resource/types"
-	resourcetypesv1 "github.com/cheqd/cheqd-node/x/resource/types/v1"
+	// resourcetypesv1 "github.com/cheqd/cheqd-node/x/resource/types/v1"
```
Comment (Eengineer1) [inline tests/upgrade/integration/v2/loader.go:13]: Marking to remove commented out lines, effective for the whole file.

File: x/did/types/tx_msg_create_diddoc_payload.go [hunk 10]
```diff
@@ -1,7 +1,7 @@
 package types
 
 import (
-	didtypesv1 "github.com/cheqd/cheqd-node/x/did/types/v1"
+	// didtypesv1 "github.com/cheqd/cheqd-node/x/did/types/v1"
```
Comment (Eengineer1) [inline x/did/types/tx_msg_create_diddoc_payload.go:4]: Marking to remove commented out lines, effective for the whole file.

File: x/resource/keeper/querier.go [hunk 11]
```diff
@@ -1,34 +1,24 @@
...
-		default:
-			err = sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unknown %s query endpoint: %s", types.ModuleName, path[0])
-		}
-
-		return res, err
-	}
-}
+// 		return res, err
+// 	}
+// }
```
Comment (Eengineer1) [inline x/resource/keeper/querier.go:24]: Marking to remove the whole file.

File: x/resource/keeper/querier_collection_resources.go [hunk 12]
```diff
@@ -1,27 +1 @@
...
-		return nil, err
-	}
-
-	bz, err := codec.MarshalJSONIndent(legacyQuerierCdc, resp)
-	if err != nil {
-		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
-	}
-
-	return bz, nil
-}
```
Comment (Eengineer1) [inline x/resource/keeper/querier_collection_resources.go:27]: Marking to remove the whole file.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]

File: app/app.go [hunk 13]
```diff
@@ -718,28 +760,37 @@ func New(
...
+	app.SetAnteHandler(anteHandler)
 	app.SetEndBlocker(app.EndBlocker)
+	// app.SetPostHandler(postHandler)
 
+	// Note: This migration is completed, we can remove these lines.
 	// Upgrade handler for v1
 	v1UpgradeHandler := app.upgradeHandlerV1(icaModule, keys[didtypes.StoreKey], keys[resourcetypes.StoreKey])
 	app.UpgradeKeeper.SetUpgradeHandler(upgradeV1.UpgradeName, v1UpgradeHandler)
-	// Upgrade handler for v2
-	app.UpgradeKeeper.SetUpgradeHandler(upgradeV2.UpgradeName, upgradeV2.CreateUpgradeHandler(app.mm, app.configurator))
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go:742]: `RegisterUpgradeHandlers` is tightly coupled with `app` struct so its better to keep the `upgradeV2.CreateUpgradeHandler` in the same app package. Is the deletion of [v2//upgrade.go](https://github.com/cheqd/cheqd-node/pull/696/files#diff-50b93f23efd1fde5990254ab02d6a6f6612cb3cb15bc7eac2e0e556c039bfa92) intended ?

File: make/proto.mk (hunk 5 above)
Comment (deepan95dev) [reply make/proto.mk]: Done

File: .github/linters/.golangci.yaml [hunk 14]
```diff
@@ -3,15 +3,18 @@ run:
 
 
 linters:
+  disable-all: true
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply .github/linters/.golangci.yaml]: Done

File: .github/linters/.golangci.yaml [hunk 15]
```diff
@@ -3,15 +3,18 @@ run:
...
   enable:
     - bodyclose
     - depguard
+    - depguard
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply .github/linters/.golangci.yaml]: Done

File: app/app.go [hunk 16]
```diff
@@ -603,16 +630,16 @@ func New(
 		ibcfee.NewAppModule(app.IBCFeeKeeper),
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go:637]: Done

File: app/app.go [hunk 17]
```diff
@@ -46,11 +57,12 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/capability"
 	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
 	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go:60]: Done

File: app/app.go [hunk 18]
```diff
@@ -131,15 +137,14 @@ var (
...
 		capability.AppModuleBasic{},
 		staking.AppModuleBasic{},
 		mint.AppModuleBasic{},
 		distr.AppModuleBasic{},
 		gov.NewAppModuleBasic([]govclient.ProposalHandler{
 			paramsclient.ProposalHandler,
-			distrclient.ProposalHandler,
 			upgradeclient.LegacyProposalHandler,
 			upgradeclient.LegacyCancelProposalHandler,
 			ibcclientclient.UpdateClientProposalHandler,
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go:153]: Done

File: app/app.go [hunk 19]
```diff
@@ -638,11 +665,11 @@ func New(
 		resourcetypes.ModuleName,
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go:672]: Done

File: app/app.go [hunk 20]
```diff
@@ -693,13 +720,28 @@ func New(
 		vestingtypes.ModuleName,
 		upgradetypes.ModuleName,
 		paramstypes.ModuleName,
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go:731]: Done

File: app/app.go [hunk 21]
```diff
@@ -638,11 +665,11 @@ func New(
...
-	app.mm.SetOrderEndBlockers(
+	app.ModuleManager.SetOrderEndBlockers(
 		crisistypes.ModuleName,
 		govtypes.ModuleName,
 		stakingtypes.ModuleName,
-		ibchost.ModuleName,
+		ibcexported.ModuleName,
 		ibctransfertypes.ModuleName,
 		capabilitytypes.ModuleName,
 		authtypes.ModuleName,
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go:683]: Done

File: app/app.go [hunk 22]
```diff
@@ -718,28 +760,37 @@ func New(
...
+	// 	tmos.Exit(err.Error())
+	// }
 
-	app.SetAnteHandler(anteHandler)
-	app.SetPostHandler(postHandler)
 	app.SetInitChainer(app.InitChainer)
 	app.SetBeginBlocker(app.BeginBlocker)
+	app.SetAnteHandler(anteHandler)
 	app.SetEndBlocker(app.EndBlocker)
+	// app.SetPostHandler(postHandler)
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go]: Restored

File: app/migrations/migration_did_protobuf.go [hunk 23]
```diff
@@ -1,176 +1,176 @@
 package migrations
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/migrations/migration_did_protobuf.go]: Done

File: app/migrations/migration_resource_protobuf.go [hunk 24]
```diff
@@ -1,80 +1,80 @@
 package migrations
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/migrations/migration_resource_protobuf.go]: Done

File: app/migrations/migrator.go [hunk 25]
```diff
@@ -2,16 +2,17 @@ package migrations
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/migrations/migrator.go]: Done

File: app/app.go [hunk 26]
```diff
@@ -718,28 +760,37 @@ func New(
...
 
+	// Note: This migration is completed, we can remove these lines.
 	// Upgrade handler for v1
 	v1UpgradeHandler := app.upgradeHandlerV1(icaModule, keys[didtypes.StoreKey], keys[resourcetypes.StoreKey])
 	app.UpgradeKeeper.SetUpgradeHandler(upgradeV1.UpgradeName, v1UpgradeHandler)
```
[the remark this answers is not in the document]
Comment (deepan95dev) [reply app/app.go]: Commented

File: tests/upgrade/integration/v1/cli/query_did.go (hunk 6 above)
Comment (deepan95dev) [reply tests/upgrade/integration/v1/cli/query_did.go]: File has been deleted

File: tests/upgrade/integration/v1/cli/query_resource.go (hunk 7 above)
Comment (deepan95dev) [reply tests/upgrade/integration/v1/cli/query_resource.go]: File has been deleted

File: tests/upgrade/integration/v1/cli/tx_did.go (hunk 8 above)
Comment (deepan95dev) [reply tests/upgrade/integration/v1/cli/tx_did.go]: File has been deleted

File: tests/upgrade/integration/v2/loader.go (hunk 9 above)
Comment (deepan95dev) [reply tests/upgrade/integration/v2/loader.go]: Done

File: x/resource/keeper/querier.go (hunk 11 above)
Comment (deepan95dev) [reply x/resource/keeper/querier.go]: Done

File: x/resource/keeper/querier_collection_resources.go (hunk 12 above)
Comment (deepan95dev) [reply x/resource/keeper/querier_collection_resources.go]: Done

**Review by Eengineer1 — APPROVED** [review]

**Pull request merged**